File: proyectos/caso4/legacy-node/src/historial.js
'use strict';

const store = require('./store');

// Filtrado por estado resuelto acá (servidor), no en el consumidor.
// Ver decisión fechada en docs/specs/spec-historial.md.
function listarHistorial({ estado } = {}) {
  const todos = store.todos();
  if (!estado) return todos;
  return todos.filter((p) => p.estado === estado);
}

module.exports = { listarHistorial };

File: proyectos/caso4/src/historial.py
"""Capacidad "historial". Reescrita desde Node.js guiada por
docs/specs/spec-historial.md. El filtro por estado se resuelve acá
(servidor), igual que en la version Node que reemplaza.
"""

from . import store


def listar_historial(estado=None):
    todos_ = store.todos()
    if not estado:
        return todos_
    return [p for p in todos_ if p["estado"] == estado]
